From 2bbe236f1d44eeed0bcc06e04117e3e32e5161a1 Mon Sep 17 00:00:00 2001
From: scx <scx.mail@gmail.com>
Date: Thu, 30 Jan 2020 02:13:00 +0100
Subject: [PATCH] Fix PDF import: replace deprecated os.popen() by
 subprocess.Popen()

The os.popen() method is obsolete since Python 2.6.
We should use the subprocess module instead.

https://docs.python.org/2/library/os.html#os.popen
https://docs.python.org/3/library/subprocess.html#replacing-os-popen-os-popen2-os-popen3
---
 src/ocrfeeder/util/lib.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/ocrfeeder/util/lib.py b/src/ocrfeeder/util/lib.py
index fa828fb..1079416 100644
--- a/src/ocrfeeder/util/lib.py
+++ b/src/ocrfeeder/util/lib.py
@@ -18,6 +18,7 @@
 ###########################################################################
 
 import os
+import subprocess
 import re
 import mimetypes
 from PIL import Image
@@ -85,7 +86,8 @@ def convertPdfToImages(pdf_file, temp_dir = '/tmp'):
                'pdf_file': pdf_path,
                'resolution': resolution,
                'size': size}
-    os.popen(command).read()
+    process = subprocess.Popen(command, shell=True, bufsize=1, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    process.communicate()
     if pdf_file != pdf_file_safe:
         try:
             os.unlink(pdf_path)
-- 
2.24.1

